internal/datastore/postgres/postgres.go

```diff
@@ -185,6 +185,25 @@ func newPostgresDatastore(
 		}
 	}
 
+	var gcPoolConfig *pgxpool.Config
+	if isPrimary {
+		// Configure a pool of size 1 for the GC connection.
+		gcPoolConfig = pgConfig.Copy()
```

Comment:
I don't think you need a new pool here, you can just `Acquire()` one from the write pool?

Reply:
But if that connection dies, will it reconnect?

Reply:
no, but I was imagining you'd just acquire one at the beginning of the gc process, which is periodic

Reply:
also - if the connection dies, you will have lost the gc lock, so you'll _need_ to reconnect explicitly and re-lock